internal/controller/storageclient_controller.go

```diff
@@ -694,10 +694,9 @@ func (r *storageClientReconcile) reconcileResource(obj client.Object, rawObject
 	}
 
 	var err error
-	if useReplace {
+	_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, obj, mutateFunc)
+	if utils.IsForbiddenError(err) {
 		err = utils.CreateOrReplace(r.ctx, r.Client, obj, mutateFunc)
```

Comment:
Does not make a lot of sense to run createOrReplace. We know create will fail. Is there a reason we cannot just run the replace code?